messaging-client-databus/src/main/java/com/inmobi/messaging/consumer/databus/DatabusConsumer.java

```diff
@@ -181,47 +206,22 @@ protected void createPartitionReaders() throws IOException {
             ((Checkpoint) currentCheckpoint).getPartitionsCheckpoint();
         LOG.info("Creating partition readers for all the collectors");
         for (String collector : getCollectors(fs, streamDir)) {
-          PartitionId id = new PartitionId(clusterName, collector);
-          PartitionCheckpoint pck = partitionsChkPoints.get(id);
-          /*
-           * Migration of checkpoint required in this case
-           * If user provides a cluster name and partition checkpoint is null
-           */
-          if (!clusterName.equals(getDefaultClusterName(i)) && pck == null) {
-            PartitionId defaultPid = new PartitionId(getDefaultClusterName(i),
-                collector);
-            pck = partitionsChkPoints.get(defaultPid);
-            /*
-             * Migrate to new checkpoint
-             */
-            ((Checkpoint) currentCheckpoint).migrateCheckpoint(pck, defaultPid, id);
-          }
-          Date partitionTimestamp = getPartitionTimestamp(id, pck);
-          LOG.debug("Creating partition " + id);
-          PartitionReaderStatsExposer collectorMetrics = new
-              CollectorReaderStatsExposer(topicName, consumerName,
-                  id.toString(), consumerNumber, fsuri);
-          addStatsExposer(collectorMetrics);
-          Path streamsLocalDir = null;
-          if (readFromLocalStream) {
-            streamsLocalDir = DatabusUtil.getStreamDir(StreamType.LOCAL,
-                rootDirs[i], topicName);
-          }
-          for (int c = 0; c < numList; c++) {
-            collectorMetrics.incrementListOps();
+          if (collectorsWithReaders.contains(collector + rootDirs[i])) {
```

Comment:
Instead of concatenating collector and rootDirs[i], better to have a separator (e.g. '_') in between. That way it will the string will be more readable if printed in future.